kazoo/protocol/connection.py

```diff
@@ -543,6 +543,8 @@ def _connect_loop(self, retry):
 
         # Check for an empty hostlist, indicating none resolved
         if len(host_ports) == 0:
+            if len(self.client.hosts) == 1:
```

Comment:
Would you be willing to explain the reason for this additional condition? Looking at the code, I would have replaced the `return STOP_CONNECTING` with a `raise ForceRetryError('No host resolved, reconnecting')` but you may have seen something I am missing.

Reply:
Thanks. I was just trying to keep the behavior to default when multiple hosts are used and if they all fail to resolve. I would also prefer to keep it simple and raise retry in all cases. Will update.